mlir-tensorrt/compiler/lib/Conversion/TensorRTToTensorRTRuntime/TensorRTToTensorRTRuntime.cpp

```diff
@@ -57,6 +57,80 @@ struct RewriteConstants : public OpRewritePattern<tensorrt::ConstantOp> {
   }
 };
 
+// Helper function to convert both CallOp and CallAllocOp
+static LogicalResult
+convertCallOp(Operation *op, MLIRContext *ctx, IRRewriter &rewriter,
+              SymbolTableCollection &symbolTable, DataFlowSolver &solver,
+              ModuleOp module,
+              ArrayRef<tensorrt::TensorRTModuleOp> trtModules) {
+  Location loc = op->getLoc();
+  ValueRange inputs;
+  ValueRange outputs;
+  func::FuncOp trtFunc;
+
+  if (auto callOp = dyn_cast<tensorrt::CallOp>(op)) {
+    trtFunc = callOp.getFuncCallee(symbolTable);
+    inputs = callOp.getInputs();
+    outputs = callOp.getOutputs();
+  } else if (auto callAllocOp = dyn_cast<tensorrt::CallAllocOp>(op)) {
+    trtFunc = callAllocOp.getFuncCallee(symbolTable);
+    inputs = callAllocOp.getInputs();
+    // CallAllocOp doesn't have outputs as operands
+  } else {
+    llvm_unreachable("unexpected type of operation. Only callOp and "
+                     "callAllocOp are supported.");
+    return failure();
+  }
+
+  solver.load<dataflow::DeadCodeAnalysis>();
+  solver.load<dataflow::SparseConstantPropagation>();
+  solver.load<TensorKindAnalysis>(symbolTable);
+  if (failed(solver.initializeAndRun(trtFunc))) {
+    trtFunc.emitError() << "failed to run TensorKindAnalysis";
+    return failure();
+  }
+
+  // Check which tensors should be host tensors.
+  SmallVector<int64_t> hostTensorArgs;
+  for (auto [idx, arg] : llvm::enumerate(trtFunc.getArguments())) {
+    const TensorKindLattice *kind = solver.lookupState<TensorKindLattice>(arg);
+    RankedTensorType rtt = cast<RankedTensorType>(arg.getType());
+    // To be conservative, we only do this if type is i32 and num elements
+    // <= 8.
+    if (kind && !kind->getValue().isUninitialized() &&
+        kind->getValue().isHostVisible() &&
+        rtt.getElementType().isInteger(32) && rtt.getNumElements() <= 8)
+      hostTensorArgs.push_back(idx);
+  }
+
+  rewriter.setInsertionPoint(op);
+
+  Value executionContext = rewriter.create<trtrt::CompileOp>(
+      loc,
+      SymbolRefAttr::get(rewriter.getStringAttr(*trtModules.front().getName()),
+                         {FlatSymbolRefAttr::get(trtFunc)}));
+  Value stream = rewriter.create<cuda::GetGlobalStreamOp>(loc, 0);
+
+  Operation *enqueueOp;
+  if (isa<tensorrt::CallOp>(op)) {
+    enqueueOp = rewriter.create<trtrt::EnqueueOp>(
+        loc, executionContext, stream, inputs, outputs,
+        /*host_tensors_args=*/hostTensorArgs.empty()
+            ? DenseI64ArrayAttr{}
+            : DenseI64ArrayAttr::get(ctx, hostTensorArgs));
+  } else {
+    enqueueOp = rewriter.create<trtrt::EnqueueAllocOp>(
+        loc, op->getResultTypes(), executionContext, stream, inputs,
+        /*host_tensors_args=*/hostTensorArgs.empty()
+            ? DenseI64ArrayAttr{}
+            : DenseI64ArrayAttr::get(ctx, hostTensorArgs));
+  }
+  rewriter.setInsertionPointAfter(enqueueOp);
```

Comment:
Changing the insertion point here shouldn't have any effect. After creation of op, the insertion point is already after the op.